lib/puppetserver/ca/action/list.rb

```diff
@@ -79,41 +88,62 @@ def run(input)
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
 
-          (all || certnames.any?) \
-            ? output_certs_by_state(requested, signed, revoked, missing)
-            : output_certs_by_state(requested)
+          (all || certnames.any?)\
+            ? output_certs_by_state(all, output_format, requested, signed, revoked, missing)
+            : output_certs_by_state(all, output_format, requested)
 
           return missing.any? \
             ? 1
             : 0
         end
 
-        def output_certs_by_state(requested, signed = [], revoked = [], missing = [])
+        def output_certs_by_state(all, output_format, requested, signed = [], revoked = [], missing = [])
+          grouped_cert = {}
           if revoked.empty? && signed.empty? && requested.empty? && missing.empty?
             @logger.inform "No certificates to list"
```

Comment:
Shouldn't there be a JSON equivalent of this message?

Reply:
I wasn't sure of that. I could add it in.  But wouldn't that just be an empty json object?

Reply:
I think an empty JSON object makes sense.

Reply:
I think if you tell the command to output JSON everything should be in JSON. I'm not sure if an empty string is valid JSON but I'm fine with an empty object.

Reply:
I just noticed the description of the PR has the output as a JSON array. Where here we talk about returning an empty object. I think we should return the same JSON type for every invocation, preferably an object.